controllers/clustergroupupgrade_controller.go

```diff
@@ -866,119 +899,95 @@ func (r *ClusterGroupUpgradeReconciler) reconcileResources(ctx context.Context,
 	return nil
 }
 
-func (r *ClusterGroupUpgradeReconciler) isClusterCompliantWithPolicy(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, clusterName string, policy *unstructured.Unstructured) bool {
-	// Get the status part of the policy.
-	statusObject := policy.Object["status"].(map[string]interface{})
-	// Get the part of the status that holds the clusters' compliance.
-	statusCompliance := statusObject["status"]
-
-	// If the status doesn't contain anything, we don't have enough information to decide, so we assume the
-	// cluster is not compliant with the policy.
-	if statusCompliance == nil {
-		return false
-	}
-
-	subStatus := statusCompliance.([]interface{})
-
-	// Loop through all the NonCompliant clusters in the policy's status.
-	for _, crtSubStatusCrt := range subStatus {
-		crtSubStatusMap := crtSubStatusCrt.(map[string]interface{})
-		// If the cluster is NonCompliant, return false.
-		if crtSubStatusMap["compliant"] == utils.StatusNonCompliant && clusterName == crtSubStatusMap["clustername"].(string) {
-			return false
-		}
-	}
-
-	return true
-}
+/*
+  getClusterComplianceWithPolicy returns the compliance of a certain cluster with a certain policy.
+  based on a policy's status structure which is below.
```

Comment:
This would be a good place to capture the information you got from ACM that the status will always include any bound clusters and be either compliant/non-compliant (reference to their API docs is even better)